gen/templates/provider.go

```diff
@@ -47,16 +47,17 @@ type FmcProvider struct {
 
 // FmcProviderModel describes the provider data model.
 type FmcProviderModel struct {
-	Username types.String `tfsdk:"username"`
-	Password types.String `tfsdk:"password"`
-	URL      types.String `tfsdk:"url"`
-	Insecure types.Bool   `tfsdk:"insecure"`
-	Retries  types.Int64  `tfsdk:"retries"`
+	Username   types.String `tfsdk:"username"`
+	Password   types.String `tfsdk:"password"`
+	URL        types.String `tfsdk:"url"`
+	Insecure   types.Bool   `tfsdk:"insecure"`
+	ReqTimeout types.String `tfsdk:"req_timeout"`
```

Comment:
Is there a specific trigger/reason to expose this in the provider config?

Reply:
If connectivity is perfect, no timeout is needed.

If connectivity is not perfect, there is almost never a case when timeout 60 seconds (default) is a benefit. What I see both in my previous experience and also in my current experience with our lab's connectivity, are short 100% blackouts, which are resolved by retries. In such case TF session takes a lot of time, as HTTP calls either complete immediately or take 60 seconds (but somehow there seem to be virtually no calls that complete in 2-59 seconds).

With our lab, yes, it was needed for me in practice, so if it's not merged I'd still probably keep the decreased timeout on local sdk code (and use `go.work` to compile it into the provider).